distributed-commandbus-jgroups/src/main/java/org/axonframework/jgroups/commandhandling/JGroupsConnector.java

```diff
@@ -470,10 +469,152 @@ protected Address resolveAddress(Member destination) {
+         *
+         * @param serializer the {@link Serializer} used to serialize command messages when they are sent between nodes
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder serializer(Serializer serializer) {
+            assertThat(serializer,
+                       Objects::nonNull,
+                       () -> new AxonConfigurationException("Serializer may not be null"));
+            this.serializer = serializer;
+            return this;
+        }
+
+        /**
+         * Sets the {@link RoutingStrategy} used to define the key based on which Command Messages are routed to their
+         * respective handler nodes. Defaults to a {@link AnnotationRoutingStrategy}, which searches for the
+         * {@link org.axonframework.commandhandling.TargetAggregateIdentifier} annotated field as the routing key.
+         *
+         * @param routingStrategy the {@link RoutingStrategy} used to define the key based on which Command Messages are
+         *                        routed to their respective handler nodes
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder routingStrategy(RoutingStrategy routingStrategy) {
+            assertThat(routingStrategy,
+                       Objects::nonNull,
+                       () -> new AxonConfigurationException("RoutingStrategy may not be null"));
+            this.routingStrategy = routingStrategy;
+            return this;
+        }
+
+        /**
+         * Sets the {@link ConsistentHashChangeListener} which is notified when a change in membership has
+         * <em>potentially</em> caused a change in the consistent hash. Defaults to a no-op solution.
+         *
+         * @param consistentHashChangeListener the {@link ConsistentHashChangeListener} which is notified when a change
+         *                                     in membership has <em>potentially</em> caused a change in the consistent
+         *                                     hash
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder consistentHashChangeListener(ConsistentHashChangeListener consistentHashChangeListener) {
+            assertThat(consistentHashChangeListener,
+                       Objects::nonNull,
+                       () -> new AxonConfigurationException("ConsistentHashChangeListener may not be null"));
+            this.consistentHashChangeListener = consistentHashChangeListener;
+            return this;
+        }
+
+        /**
+         * Initializes a {@link JGroupsConnector} as specified through this Builder.
+         *
+         * @return a {@link JGroupsConnector} as specified through this Builder
+         */
+        public JGroupsConnector build() {
+            return new JGroupsConnector(this);
+        }
+
+        private void validate() {
+            assertThat(localSegment,
```

Comment:
Extraction to a separate method could help here.

Reply:
Replaced for the `BuilderUtils` functions.
What do you think?